geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -172,29 +275,112 @@ def before_create(table, bind, **kw):
                     col.type, "spatial_index", False
                 ):
                     table.info["_after_create_indexes"].append(idx)
-    _setup_dummy_type(table, gis_cols)
+
+    if db_is_gpkg:
+        if len(gis_cols) > 1:
+            raise ValueError(
+                "Only one geometry column is allowed for a table stored in a GeoPackage."
+            )
+        elif len(gis_cols) == 1:
+            col = gis_cols[0]
+            srid = col.type.srid
+
+            if col.type.geometry_type is None:
+                col.type.geometry_type = "GEOMETRY"
+
+            # Add the SRID of the table in 'gpkg_spatial_ref_sys' if this table exists
+            if (
+                bind.execute(text("""PRAGMA main.table_info("gpkg_spatial_ref_sys");""")).fetchall()
+                and not bind.execute(
+                    text(
+                        """SELECT COUNT(*) FROM gpkg_spatial_ref_sys WHERE srs_id = {};""".format(
+                            srid
+                        )
+                    )
+                ).scalar()
+            ):
+                bind.execute(text("SELECT gpkgInsertEpsgSRID({})".format(srid)))
+            # table.columns = table.info.pop("_saved_columns")
+        _setup_dummy_type(table, gis_cols, gpkg=True)
+    else:
+        _setup_dummy_type(table, gis_cols, gpkg=False)
 
 
 def after_create(table, bind, **kw):
     """Handle spatial indexes during the after_create event."""
     dialect = bind.dialect
     dialect_name = dialect.name
-
-    table.columns = table.info.pop("_saved_columns")
+    db_is_gpkg = is_gpkg(bind)
+
+    if db_is_gpkg:
+        for col in table.columns:
+            # Add the managed Geometry columns with gpkgAddGeometryColumn()
+            if _check_spatial_type(col.type, Geometry, dialect) and check_management(
+                col, dialect_name
+            ):
+                col.type = col._actual_type
+                del col._actual_type
+                dimension = get_col_dim(col)
+                has_z = "Z" in dimension
+                has_m = "M" in dimension
+
+                bind.execute(
```

Comment:
There is always an "SQL INJECTION" alarm going off in my head if I see string formatting on raw sql. It's probably fine here, but still, it's easy to do it like SQLAlchemy advertises. Unless I am missing something.

```
t = text("SELECT * FROM users WHERE id=:user_id")
result = connection.execute(t, user_id=12)
```

Reply:
Well, in our case we only use attributes that come from SQLAlchemy objects (e.g. `table.name\, `col.name`, etc.). So it's hard for SQL injections to come up to here. But I guess it's not impossible and any way perfs are not really important  for administration functions so let's use parameters to be sure.